File: app/src/main/java/name/boyle/chris/sgtpuzzles/RestartActivity.java
package name.boyle.chris.sgtpuzzles;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

@SuppressWarnings("WeakerAccess")  // used by manifest
public class RestartActivity extends Activity
{
	@Override
	protected void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);
		startActivity(new Intent(this, GamePlay.class));
		finish();
	}
}

File: app/src/main/kotlin/name/boyle/chris/sgtpuzzles/RestartActivity.kt
package name.boyle.chris.sgtpuzzles

import android.app.Activity
import android.content.Intent
import android.os.Bundle

class RestartActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        startActivity(Intent(this, GamePlay::class.java))
        finish()
    }
}